packages/attribute-registry-consumer/src/attributeRegistryConsumerService.ts

```diff
@@ -0,0 +1,34 @@
+import { match } from "ts-pattern";
+import {
+  logger,
+  consumerConfig,
+  ReadModelRepository,
+} from "pagopa-interop-commons";
+import { EventEnvelope } from "./model/models.js";
+import { fromAttributeV1 } from "./model/converter.js";
+
+const { attributes } = ReadModelRepository.init(consumerConfig());
+
+export async function handleMessage(message: EventEnvelope): Promise<void> {
+  logger.info(message);
+  await match(message)
+    .with({ type: "AttributeAdded" }, async (msg) => {
+      await attributes.updateOne(
+        {
+          "data.id": msg.stream_id,
+        },
+        {
+          $setOnInsert: {
+            data: msg.data.attribute
```

Comment:
From a type declaration point of view, `protobuf-ts` transforms every `message` field into a `T | undefined` field so it depends on how the `process` creates the event.

Reply:
uhm why is that? in the [proto file](https://github.com/pagopa/interop-be-monorepo/pull/83/files/22bfab8ab78a6dfb2a0ecd924c1540f90f905bc3#diff-3ab22be2aa8668ded2719855ff8dfbc04905bfc454a92d517166fcdf20230406R8) the field is flagged as `required`

Reply:
The `required` field should not be used (https://protobuf.dev/programming-guides/proto2/#field-labels) because is problematic.
In `protobuf 3` the `required` where removed. The typescript library that we use doesn't implement the `required` field for this reason.

Reply:
thanks!